RiotSwiftUI/Modules/Authentication/Common/Service/MatrixSDK/AuthenticationService.swift

```diff
@@ -138,9 +138,7 @@ class AuthenticationService: NSObject {
         loginWizard = nil
         registrationWizard = nil
         
-        // The previously used homeserver is re-used as `startFlow` will be called again a replace it anyway.
-        let address = state.homeserver.addressFromUser ?? state.homeserver.address
-        self.state = AuthenticationState(flow: .login, homeserverAddress: address)
+        self.state = AuthenticationState(flow: .login, homeserverAddress: state.homeserver.address)
```

Comment:
Would it be possible to revert this? It is intentional that if we call `startFlow` using the existing address, that it should be the address that the user entered e.g. `matrix.org` and not the resolved address e.g. `matrix-client.matrix.org`.